WordPress/wordpress-playground
# Offline support

## Motivation for the change, related issues

Most of Playgrounds load time is waiting for files to download. 
If the user reloads Playground they are waiting for the same files to download.

This PR improves Playground load time by caching WP, PHP, and the SQLite plugin in the browser.

## Implementation details

The PR uses browser cache storage to cache fetch responses and uses the request URL as the cache key.
Because WP, PHP, and the SQLite plugin change URLs when they are rebuilt the cache will be busted and the correct version will be downloaded.



## Testing Instructions (or ideally a Blueprint)

- Checkout this branch
- Open Playground in Chrome
- In dev tools > Application clear all cache storage
- See that `wp-6.5.zip` was downloade
- Reload and see that it wasn't downloaded
